src/uniprotkb/components/protein-data-views/FreeTextView.tsx

```diff
@@ -20,50 +21,82 @@ const needsNewLineRE = /^\).\s+/;
 
 const getEntryPathForCitation = getEntryPathFor(Namespace.citations);
 
+export const ACCommentView = ({ string }: { string: string }) => {
+  // This replaces any occurrences of "AC <accession>" with "AC "<link to the accession>
+  const acRE = new RegExp(`(AC ${uniProtKBAccessionRegEx.source})`, 'i');
+  return (
+    <>
+      {string.split(acRE).map((part, index, { length }) => {
+        const accession = part.match(uniProtKBAccessionRegEx)?.[0];
+        return acRE.test(part) && accession ? (
+          // eslint-disable-next-line react/no-array-index-key
+          <Fragment key={index}>
+            {`AC `}
+            <Link to={getEntryPath(Namespace.uniprotkb, accession)}>
+              {accession}
+            </Link>
+          </Fragment>
+        ) : (
+          // eslint-disable-next-line react/no-array-index-key
+          <Fragment key={index}>
+            {index + 1 === length && !part.endsWith('.') ? `${part}.` : part}
+          </Fragment>
```

Comment:
I believe you might not need a Fragment nor a key when you return a string. If that is the case you could remove that here.
That being said, you'll get eslint complaining about ternary operator inside a ternary operator.
I would do multiple returns in different if conditions so that it's clearer to see what is returned under which conditions.

Reply:
Done